### ISSUE TITLE
vcftools convert genotype of '.|.' to '.'

### ISSUE THREAD

**Issue by ntakebay** [vcftools/vcftools#131]
With GenotypeGVCFs of GATK 4.1.1.0 creates genotype (GT) like '.|.'  vcftools (v 0.1.16) seems to convert this into '.'  For example,
.|.:0,0:.:.:0|1:15530_TTG_T:.:15530
becomes
.:0,0:.:.:0|1:15530_TTG_T:.:15530

Then there is also weird conversion happening:

0/0:13,0:13:39:.:.:0,39,547	./.:0,0:0:.:.:.:0,0,0	0/0:1,0:1:3:.:.:0,3,32	0/0:9,0:9:24:.:.:0,24,360
Qmera81.spades_2325	15525	.	A	G	64.71	.	AC=1;AF=0.125;AN=8;BaseQRankSum=-6.740e-01;DP=27;ExcessHet=3.0103;FS=4.771;MLEAC=1;MLEAF=0.125;MQ=45.89;MQRankSum=-1.383e+00;QD=16.18;ReadPosRankSum=0.431;SOR=2.209	GT:AD:DP:GQ:PGT:PID:PL:PS	0|1:2,2:4:78:0|1:15504_TA_T:78,0,78:15504	

becomes
 Qmera81.spades_2325	15525	.	A	G	64.71	.	AC=1;AF=0.125;AN=8;BaseQRankSum=-6.740e-01;DP=27;ExcessHet=3.0103;FS=4.771;MLEAC=1;MLEAF=0.125;MQ=45.89;MQRankSum=-1.383e+00;QD=16.18;ReadPosRankSum=0.431;SOR=2.209	GT:AD:DP:GQ:PGT:PID:PL:PS	0|1:2,2:4:78:0|1:15504_TA_T:78,0,78:15504	0/0:13,0:13:39:.:.:0,39,547:.	./.:0,0:0:.:.:.:0,0,0:.	0/0:1,0:1:3:.:.:0,3,32:.	0/0:9,0:9:24:.:.:0,24,360:.

Basically vcftools is adding additional field at the end  of each genotype with '.' as the value.

I'm attaching a test file which causes this problem (only 2 sites extracted from the full vcf), and I used the following command.

vcftools --vcf temp.vcf --recode --recode-INFO-all  --out tempOut

[temp.vcf.txt](https://github.com/vcftools/vcftools/files/3114601/temp.vcf.txt)

**Comment by Dario-Galanti** [comment]
Hello, I run into the exact same issue. After I filter and --recode my multisample vcf file, I get all ".|." transformed into "." This messes up my next step which is imputation using BEAGLE, which throws an error when not finding any allele separator in the GT field.
I read the issue #13 above but it does not seem to offer a solution as filtering out good positions just because very few samples run into this issues seems a pity.
Has anyone found any solution to this in the meantime?

**Comment by Dario-Galanti** [comment]
Solution found! I just manually changed back "." in the genotype field with the original ".|." here is the command (the compressed version requires sometools):
cat filtered.vcf | sed $'s/\t\.:/\t\.|\.:/g' > filtered_ready.vcf #Uncompressed version
zcat filtered.vcf.gz | sed $'s/\t\.:/\t\.|\.:/g' | bgzip -c > filtered_ready.vcf.gz #Compressed version
